server/aesdsocket.c

```diff
@@ -92,43 +179,40 @@ int main(int argc, char *argv[]) {
 
     syslog(LOG_INFO, "Server listening on port %d", PORT);
 
+    // Start timestamp thread
+    pthread_t time_thread;
+    pthread_create(&time_thread, NULL, timestamp_thread, NULL);
+
+    LIST_INIT(&thread_head);
+
     while (1) {
-        client_socket = accept(server_socket, (struct sockaddr *)&client_addr, &client_addr_len);
+        int client_socket = accept(server_socket, (struct sockaddr *)&client_addr, &client_addr_len);
         if (client_socket == -1) {
             syslog(LOG_ERR, "Accept failed: %s", strerror(errno));
             continue;
         }
 
         syslog(LOG_INFO, "Accepted connection from %s", inet_ntoa(client_addr.sin_addr));
 
-        FILE *file = fopen(FILE_PATH, "a+");
-        if (!file) {
-            syslog(LOG_ERR, "Failed to open file: %s", strerror(errno));
+        // Allocate memory for new thread
+        client_thread_t *new_thread = malloc(sizeof(client_thread_t));
+        if (!new_thread) {
+            syslog(LOG_ERR, "Malloc failed");
             close(client_socket);
             continue;
         }
 
-        ssize_t bytes_received;
-        while ((bytes_received = recv(client_socket, buffer, BUFFER_SIZE, 0)) > 0) {
-            fwrite(buffer, 1, bytes_received, file);
-            fflush(file);
-
-            if (memchr(buffer, '\n', bytes_received)) {
-                rewind(file);
-                while ((bytes_received = fread(buffer, 1, BUFFER_SIZE, file)) > 0) {
-                    send(client_socket, buffer, bytes_received, 0);
-                }
-                break;
-            }
-        }
+        new_thread->client_socket = client_socket;
+        LIST_INSERT_HEAD(&thread_head, new_thread, entries);
```

Comment:
What is a potential issue with manipulating the link list both inside and outside the `client_handler` thread?